lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php

```diff
@@ -82,16 +82,16 @@ public function execute(InputInterface $input, OutputInterface $output)
             $path = is_bool($path) ? getcwd() : $path;
             $version->writeSqlFile($path, $direction);
         } else {
-            $noInteraction = $input->getOption('no-interaction') ? true : false;
-            if ($noInteraction === true) {
-                $version->execute($direction, $input->getOption('dry-run') ? true : false);
+            if ($input->getOption('no-interaction')) {
```

Comment:
``$input->isInteractive()`` would be even cleaner actually (the ``no-interaction`` option is an Application-level option, but the Application has other ways to guess that the input cannot be interactive in some case).